packages/loader/container-loader/src/container.ts

```diff
@@ -932,19 +932,24 @@ export class Container
 				? summaryTree
 				: combineAppAndProtocolSummary(summaryTree, this.captureProtocolSummary());
 
-		// Whether the combined summary tree has been forced on by either the loader option or the monitoring context.
+		// Whether the combined summary tree has been forced on by either the loader option or the monitoring context or supportedFeatures flag by the service.
 		// Even if not forced on via this flag, combined summaries may still be enabled by service policy.
-		const forceEnableSummarizeProtocolTree =
+		const shouldSummarizeProtocolTree =
 			this.mc.config.getBoolean("Fluid.Container.summarizeProtocolTree2") ??
-			options.summarizeProtocolTree;
+			options.summarizeProtocolTree ??
+			(typeof this._deltaManager.connectionManager.connectionDetails?.supportedFeatures
```

Comment:
this could just be:
options.summarizeProtocolTree ?? 
    this._deltaManager.connectionManager.connectionDetails?.supportedFeatures.?.enable_single_commit_summary ?? false.

Reply:
First, this is Container constructor, we have no connection to ordering service when we get here, so the answer will always be - there is no connectionDetails.
But more important - we should expose this capability ("should we produce single-commit summary") in a type-safe matter from the driver, somehow.